stock_request_non_internal/models/stock_request_abstract.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class StockRequest(models.AbstractModel):
+    _inherit = "stock.request.abstract"
+
+    warehouse_id = fields.Many2one(required=False)
```

Comment:
Why are you changing this ?

Reply:
Ok, just read the readme. The problem is that this changes the default behaviour too.
Maybe is there a more elegant way to do so? Introducing a 'type' maybe.

Reply:
I'll try to find a way. It is difficult because virtual locations are not linked to any company, and any warehouse. The ultimate purpose is to post expenses when moving products to those virtual locations. That is only possible if the locations are out of the company.

Maybe the same can be achieved by changing the constrains and the domains on the fields. I'll try.